app/controllers/api/object_definitions_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Api
+  class ObjectDefinitionsController < BaseController
+    def show
+      object = fetch_object_definition(params['c_id'])
+      render_resource :object_definitions, object
+    end
+
+    private
+
+    def fetch_object_definition(id)
+      klass = collection_class(:object_definitions)
+      obj = klass.find_by(:name => id) || klass.find(id)
+      raise ActiveRecord::RecordNotFound unless obj
```

Comment:
@abellotti not quite sure I understand. here is a resource:
```
#<GenericObjectDefinition:0x007feabcedd6c8
  id: 10000000000010,
  name: "foo",
  description: nil,
  properties: {:attributes=>{}, :associations=>{}, :methods=>[]},
  created_at: Thu, 06 Jul 2017 17:05:24 UTC +00:00,
  updated_at: Thu, 06 Jul 2017 17:05:24 UTC +00:00>
```
added in the filter on `name` based of what is defined here: https://www.pivotaltracker.com/story/show/121850669